### ISSUE TITLE
Global and hierarchical state

### ISSUE THREAD

**Issue by masihyeganeh** [kellerkindt/asn1rs#69]
I love this rustic implementation of parser using `TryFrom` but it needs a global state e.g. to support `IMPLICIT`/`EXPLICIT`/`AUTOMATIC` `TAGS` that will affect all sub structures

**Comment by kellerkindt** [comment]
Can you link and/or explain what the differences is between `IMPLICIT`/`EXPLICIT`/`AUTOMATIC TAGS` is? With `uPER` this didn't really matter until now

**Comment by masihyeganeh** [comment]
Yeah, as I said before, there is a  good post in let's encrypt for that:
[https://letsencrypt.org/docs/a-warm-welcome-to-asn1-and-der/#explicit-vs-implicit](https://letsencrypt.org/docs/a-warm-welcome-to-asn1-and-der/#explicit-vs-implicit)

**Comment by masihyeganeh** [comment]
Hey @kellerkindt,
Could you find spare time to look at this?

**Comment by kellerkindt** [comment]
TBH, I did not. I don't have a lot of time left over to work on this at the moment. But I want to at some point - maybe at or until chirstmas I will have some spare time for this. Regarding the state tracking and from what I have seen while glossing over the linked article, wouldn't it fit better in the `Model` type with an additional field?

**Comment by kellerkindt** [comment]
Cross referencing for #34
